### PR TITLE
Add new resource opsgenie_team_membership, refs #65

### PR DESCRIPTION
**Pull request by arnisoph**
Signed-off-by: Arnold Bechtoldt <arnold.bechtoldt@inovex.de>

### CODE REVIEW

File: opsgenie/resource_opsgenie_team_membership.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/opsgenie/opsgenie-go-sdk-v2/team"
+	"log"
+	"strings"
+)
+
+func resourceOpsGenieTeamMembership() *schema.Resource { //TODO encode the e-mail addrs (e.g. because of +)? https://github.com/opsgenie/opsgenie-go-sdk-v2/issues/62
```
Comment (arnisoph) [inline opsgenie/resource_opsgenie_team_membership.go:12]: might not become an issue at all. see https://github.com/opsgenie/opsgenie-go-sdk-v2/issues/62

File: opsgenie/resource_opsgenie_team_membership.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+	}
+
+	// add member to team
+	_, err = client.AddMember(context.Background(), &team.AddTeamMemberRequest{
+		TeamIdentifierType:  team.Id,
+		TeamIdentifierValue: teamID,
+		User: team.User{
+			ID: userID,
+		},
+		Role: role, //TODO lowercase?
```
Comment (arnisoph) [inline opsgenie/resource_opsgenie_team_membership.go]: I did not really understand why (global) user and team roles have different casing. Should we do some user input sanitization here?

File: opsgenie/resource_opsgenie_team_membership.go [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+	client, err := team.NewClient(meta.(*OpsgenieClient).client.Config)
+	if err != nil {
+		return err
+	}
+
+	_, err = client.RemoveMember(context.Background(), &team.RemoveTeamMemberRequest{
+		TeamIdentifierType:    team.Id,
+		TeamIdentifierValue:   teamID,
+		MemberIdentifierType:  team.Id,
```
Comment (arnisoph) [inline opsgenie/resource_opsgenie_team_membership.go:123]: sounds a bit weird but works

File: opsgenie/resource_opsgenie_team_membership.go (hunk 2 above)
Comment (ffahri) [reply opsgenie/resource_opsgenie_team_membership.go]: Customers can create a custom role for their OG accounts. But why do you think we use lowercase

File: opsgenie/resource_opsgenie_user.go [hunk 4]
```diff
@@ -53,6 +53,8 @@ func resourceOpsGenieUser() *schema.Resource {
 }
 
 func resourceOpsGenieUserCreate(d *schema.ResourceData, meta interface{}) error {
+	//TODO OGS-1629: Atlassian systems reset full_name after a certain time after creating a new OpsGenie user.
+	// This may lead to unexpected behaviour, e.g. when running a subsequent "tf apply" or executing our acceptance tests
```
Comment (ffahri) [inline opsgenie/resource_opsgenie_user.go:57]: we need to think this through, i'll watch OGS-1629

File: opsgenie/resource_opsgenie_team_membership.go (hunk 2 above)
Comment (arnisoph) [reply opsgenie/resource_opsgenie_team_membership.go]: standard team roles are lowercase. System user roles start with an uppercase. A little bit of a mess... :D

Comment (arnisoph) [reply opsgenie/resource_opsgenie_team_membership.go]: will remove the comment in a min

File: opsgenie/resource_opsgenie_team_membership.go [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+	"github.com/opsgenie/opsgenie-go-sdk-v2/team"
+	"log"
+	"strings"
+)
+
+func resourceOpsGenieTeamMembership() *schema.Resource { //TODO encode the e-mail addrs (e.g. because of +)? https://github.com/opsgenie/opsgenie-go-sdk-v2/issues/62
+	return &schema.Resource{
+		Create: resourceOpsGenieTeamMembershipCreate,
+		Read:   handleNonExistentResource(resourceOpsGenieTeamMembershipRead),
+		//Update: resourceOpsGenieTeamMembershipUpdate, // requires https://github.com/opsgenie/opsgenie-go-sdk-v2/issues/59
```
Comment (ffahri) [inline opsgenie/resource_opsgenie_team_membership.go:16]: I didn't quite understand the issue in go-sdk can you describe the issue

File: opsgenie/resource_opsgenie_team_membershop_test.go [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+
+func testAccOpsGenieTeamMembership_basic(rString string) string {
+	return fmt.Sprintf(`
+resource "opsgenie_team" "monkeys" {
+  name           = "monkeys-%s"
+  description    = "They exist."
+  ignore_members = true
+}
+resource "opsgenie_user" "kong" {
+  username   = "kong-%s@inovex.de"
```
Comment (ffahri) [inline opsgenie/resource_opsgenie_team_membershop_test.go]: i highly suggest change email to example tlds. Your smtp server will get a lot of requests

File: website/docs/r/team_membership.html.markdown [hunk 7]
```diff
@@ -0,0 +1,64 @@
+---
```
Comment (ffahri) [inline website/docs/r/team_membership.html.markdown:1]: one more thing, 
can you add this to opsgenie.erb file as well terraform website shows urls based on that file

Comment (arnisoph) [reply website/docs/r/team_membership.html.markdown:1]: ouch. thx! will change

File: opsgenie/resource_opsgenie_team_membershop_test.go (hunk 6 above)
Comment (arnisoph) [reply opsgenie/resource_opsgenie_team_membershop_test.go]: ouch. thx! will change

File: opsgenie/resource_opsgenie_team_membership.go (hunk 5 above)
Comment (arnisoph) [reply opsgenie/resource_opsgenie_team_membership.go:16]: There's no way to update a member ship with go-sdk, that's why don't support it here too. Updating a membership would only involve changing a role without removing+adding the user from the team.

**Pull request closed without merging**